### PR TITLE
[Enhancement] Support delete sql for cloud native primary key table with sort key

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #issue

refer to: https://github.com/StarRocks/starrocks/pull/25644

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 5 / 13 (38.46%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/delta_writer.cpp|5|13|38.46%|[324, 325, 326, 327, 328, 329, 330, 383]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 5 / 13 (38.46%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/delta_writer.cpp|5|13|38.46%|[324, 325, 326, 327, 328, 329, 330, 383]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 5 / 13 (38.46%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/delta_writer.cpp|4|12|33.33%|[323, 324, 325, 326, 327, 328, 329, 382]|
|:large_blue_circle:|src/storage/lake/rowset_update_state.cpp|1|1|100.00%|[]|

File: be/src/storage/delta_writer.cpp [hunk 1]
```diff
@@ -528,7 +528,6 @@ void DeltaWriter::_reset_mem_table() {
                                                 _mem_table_sink.get(), "", _mem_tracker);
     }
     _mem_table->set_write_buffer_row(_memtable_buffer_row);
-    _mem_table->set_partial_schema_with_sort_key(_partial_schema_with_sort_key);
```
Comment (luohaha) [inline be/src/storage/delta_writer.cpp:531]: Why remove this?

**Review by luohaha — APPROVED** [review]

File: be/src/storage/delta_writer.cpp (hunk 1 above)
Comment (TszKitLo40) [reply be/src/storage/delta_writer.cpp:531]: > Why remove this?

It is also useless in shared-nothing.

**Review by sevev — APPROVED** [review]

File: be/src/storage/lake/delta_writer.cpp [hunk 2]
```diff
@@ -315,8 +318,24 @@ Status DeltaWriterImpl::open() {
...
 }
 
+Status DeltaWriterImpl::_check_partial_update_with_sort_key(const Chunk& chunk) {
+    if (_partial_schema_with_sort_key && _slots != nullptr && _slots->back()->col_name() == "__op") {
+        size_t op_column_id = chunk.num_columns() - 1;
+        auto& op_column = chunk.get_column_by_index(op_column_id);
+        auto* ops = reinterpret_cast<const uint8_t*>(op_column->raw_data());
+        for (size_t i = 0; i < chunk.num_rows(); i++) {
+            if (ops[i] == TOpType::UPSERT) {
+                LOG(WARNING) << "table with sort key do not support partial update";
```
Comment (wyb) [inline be/src/storage/lake/delta_writer.cpp:328]: should update the message, only `upsert` is not supported

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**